src/js/api/ExercisesService.js

```diff
@@ -77,31 +49,17 @@ export default class ExercisesService {
     }
   }
 
-  /**
-   * Fetches exercises based on the current filters, name, and pagination.
-   * @returns {Promise<Object>} The data of exercises fetched from the API.
-   * @throws {Error} Throws an error if the request fails.
-   */
-  async getExercises() {
+  async getExercises(params) {
     try {
       const { data } = await this.#axios.get(
-        `${this.#basePath}?${
-          this.#filter && this.#name ? `${this.#filter}=${this.#name}&` : ''
-        }&page=${this.#page}&limit=10`
+        `${this.#basePath}?${params.toString()}`
```

Comment:
`const { data } = await this.#axios.get(this.#basePath, params) `

Reply:
I tried this approach, but it wasn't working correctly